[Bug] Permission denied (os error 13) when launching on macOS using java downloaded by SJMCL

### Description

When launching the game on macOS using java downloaded by SJMCL, the launcher fails with the following error:
```
[webview:error][ERROR] Invoke launchGame failed: Failed to launch game - Permission denied (os error 13)
```
The launch command shows that the launcher is trying to use:
```
~/Library/Application Support/SJMCL/runtime/java-21/jre.bundle/Contents/Home/bin/java
```
Running this manually in the terminal gives the same permission error:
```
zsh: permission denied: ~/Library/Application Support/SJMCL/runtime/java-21/jre.bundle/Contents/Home/bin/java
```

### OS

macOS Sequoia 15.6

### SJMCL Version (Optional)

dev

### Reproduction

Download java runtime using SJMCL, then launch the game.

### Expected Behavior

The launcher should start Minecraft normally using the bundled Java runtime.

### Environment (Optional)

_No response_

### Additional context (Optional)

_No response_

--- a/src-tauri/src/launch/commands.rs
+++ b/src-tauri/src/launch/commands.rs
@@ -25,7 +25,7 @@ use crate::launcher_config::models::{
 use crate::resource::helpers::misc::get_source_priority_list;
 use crate::storage::load_json_async;
 use crate::tasks::commands::schedule_progressive_task_group;
-use crate::utils::fs::create_zip_from_dirs;
+use crate::utils::fs::{create_zip_from_dirs, manage_permissions_unix, PermissionOperation};
 use crate::utils::logging::get_launcher_log_path;
 use crate::utils::shell::{execute_command_line, split_command_line};
 use crate::utils::window::create_webview_window;
@@ -79,6 +79,13 @@ pub async fn select_suitable_jre(
       .major_version,
   )
   .await?;
+
+  // ensure execute permissions (for Linux and macOS)
+  manage_permissions_unix(
+    &selected_java.exec_path,
+    0o111,
+    PermissionOperation::Upgrade,
+  )?;
 
   let timestamp = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
   let mut launching = launching_queue_state.lock()?;

--- a/src-tauri/src/utils/fs.rs
+++ b/src-tauri/src/utils/fs.rs
@@ -487,7 +487,7 @@ where
       }
       PermissionOperation::Downgrade => {
         // "downgrade" operation: only remove the permissions if they are present
-        if current_mode & perm_mask == perm_mask {
+        if current_mode & perm_mask != 0 {
           let new_mode = current_mode & !perm_mask; // Clear the respective permission
           permissions.set_mode(new_mode);
           fs::set_permissions(&path, permissions)?;

